cspatients/views.py

```diff
@@ -103,28 +94,23 @@ def rp_event(request):
     # Takes in the information from Rapid Pro
     status_code = 405
     if request.method == "POST":
-        status_code = 400
         if request.GET.get('secret') == "momkhulu":
-            patient = PatientSerializer(data=get_patient_dict(request.body))
-            patiententry = PatientEntrySerializer(
-                data=get_patiententry_dict(request.body)
-            )
-            if patient.is_valid():
-                patient.save()
-            if patiententry.is_valid():
-                patiententry.save()
-                status_code = 201
+            status_code = save_model(request.POST)
+            if status_code == 201:
                 send_consumers_table()
     return HttpResponse(status=status_code)
 
 
 @csrf_exempt
 def patientexists(request):
+    """
+        Returns 200 if Patient exists and 400 if Patient Does Not Exist
+    """
     status_code = 405
     if request.method == "POST":
         try:
             Patient.objects.get(
```

Comment:
Take a look at https://docs.djangoproject.com/en/2.0/topics/http/shortcuts/#get-object-or-404
That method might help you remove some of the overhead you're doing in some places